README.md

```diff
@@ -394,6 +380,41 @@ const result = await vc.verify({
 // {valid: true}
 ```
 
+#### `challenge` parameter
+
+Verifiable Presentations are typically used for authentication purposes.
+A `challenge` param (similar to a `nonce` in OAuth2/OpenID Connect) is provided
+by the party that's receiving the VP, and serves to prevent presentation replay
+attacks. The workflow is:
+
+1. Receiving party asks for the VerifiablePresentation, and provides a
+  `challenge` parameter.
+2. The client code creating the VP passes in that challenge (from the requesting
+  party), and it gets included in the VP.
+3. The client code passes the VP to the receiving party, which then checks to
+  make sure the `challenge` is the same as the one it provided in the request
+  in 1).
+
+### Verifying a Verifiable Credential
+
+For most situations, you will only need to verify a Verifiable Presentation,
+since Verifiable Credentials that are not wrapped in a VP are a rare occurrence.
+
+Pre-requisites:
+
+* If you're using a custom `@context`, make sure it's resolvable
+* You're using the correct public key and corresponding suite
+* Your Controller Document is reachable via a `documentLoader`
```

Comment:
Hmm, I think this whole section needs a bit of a rewrite. I started to rework it through suggestions here:

```suggestion
* If you're using a custom `@context`, make sure it's resolvable,
  providing a custom `documentLoader` as needed
* You're passing the cryptographic suite(s) your application supports
  (be prepared to handle errors if the VC does not use a matching suite)
* You have specified a specific verification method (e.g., public key) that
  your application will accept via a suite *or* you provide a custom
  `documentLoader` that can resolve acceptable verification methods
* Controller Document is reachable via a `documentLoader`
```

But it got unwieldy. I think you need to say that you should provide a `documentLoader` that can:

* Resolve any acceptable JSON-LD contexts
* Resolve any acceptable verification methods
* Resolve any acceptable controller documents for those verification methods

It's strange to talk about providing a public key to the suite or to "pick the right suite" for the VC. That's not generally how verification software should be written. Instead, you should be specifying the suites the application supports and a document loader that will only load the things that are acceptable to the application. Then you pass the VC and see if it meets your criteria or not.

Reply:
I agree - I figured we can come back and expand the docs, in a separate pr.